1875_Mp3_Player_Clone.js

```diff
@@ -0,0 +1,316 @@
+    if(getSongIdIndex(id) === undefined) throw("song id not found in list");
+    player.songs.splice(getSongIdIndex(id),1); //removed song from songs list. if id exists.
+    for(let i=0; i<player.playlists.length;i++) //go over the playlists.
+    {
+      if(player.playlists[i].songs.includes(id))  player.playlists[i].songs.splice(player.playlists[i].songs.indexOf(id), 1); // removes the id from the playlist, if exists.
+    }
+  }
+  
+  function addSong(title, album, artist, duration, id) {
+    let elementID= freeSongID(); //assign it a unused id identifier!
+    if((id !== undefined)&&(typeOf(getSongIdIndex(id)) !== 'number')) player["songs"].push({id: id, title: title, album: album, artist: artist, duration: mmtoss(duration)});
+    else player["songs"].push({id: elementID , title: title, album: album, artist: artist, duration: mmtoss(duration)});
+    return elementID;
+  }
+
+  //console.log(addSong("love","kaki","kiki","3:15"))
+  //console.log(player.songs)
+
+  function mmtoss(minutes) // translate mm:ss to ss 
+  {
+   const mmtoss = minutes.split(':'); //seperates between minutes and seconds and puts it inside a new array.
+   return parseInt(mmtoss[0]) * 60 + parseInt(mmtoss[1]); //converts to seconds. and adds up.
+  }
+
+  function freeSongID(){ // func return the free id which not yet registered to any song in the list.
+    let free=1;
+    
+    while(typeof(getSongIdIndex(free)) === 'number'){
+      free++;
+    } return free;
+  }
+
+  
+  function removePlaylist(id) {
+    
+    if(getPlaylistIdIndex(id) === undefined) throw("mannn, there is no playlist like this exist!!! Dont you see?!?"); // checks if the functions returns something it means the id exists in playlist. else, it's an error.
+
+    let i=0;
+    while(i < player.playlists.length)
+    {
+      if(player.playlists[i].id === id) {player.playlists.splice(i,1); break;}
+      i++;
+    }
+  
+  }
+
+function freeListID(){ // func return the free id which not yet registered to any playlist in the playlists.
+    let free=1;
+    for(let i in player["playlists"])
+    {
+      if(player["playlists"][i].id !== free) return free; 
+      ++free;
+    }
+    return free;
+  }
+  
+  function createPlaylist(name, id) {
+    let elementID = freeListID();
+    player.playlists.push({id: id || elementID, name: name, songs: []}) //if id inserted then it gettin assigned. else, a free one gettin assigned.
+    return elementID;
+  }
+  
+  function playPlaylist(id) {
+   if(getPlaylistIdIndex(id) === undefined) throw("playlist like this, leads to, error like this, leads to, try again...!");
+   let playlistIndex = getPlaylistIdIndex(id); //index of the playlist with the specific ID in the playlits array
+   player.playlists[playlistIndex].songs.forEach(element => playSong(element));
+  }
+  
+  
+  function editPlaylist(playlistId, songId) {
+    let playlistIndex = getPlaylistIdIndex(playlistId); //index of the playlist with the specific ID in the playlits array
```

Comment:
why not const?

Reply:
i didnt want to mess wiht const cuz it was beggining of our learning so didnt knw exactly when to use it.. but now im utryna use only consts from now on unless i get an error